src/lib/EDSL.mli

```diff
@@ -23,31 +37,76 @@ val nop : unit t
 val if_then_else :
   bool t -> unit t -> unit t -> unit t
 val if_then : bool t -> unit t -> unit t
+
+val if_seq:
+  t:unit t list ->
+  ?e:unit t list ->
+  bool t ->
+  unit t
+(** [condition c ~t ~e] is an alternate API for {!if_then_else} (when
+    [?e] is provided) or {!if_then} (otherwise) that assumes “then”
+    and “else” bodies to be lists for {!seq} construct. *)
+
 val seq : unit t list -> unit t
 val not : bool t -> bool t
-val printf : ('a, unit, string, unit t) format4 -> 'a
-val file_exists : string -> bool t
+
+val file_exists : string t -> bool t
+
+(** {3 Switch Statements } *)
+
 val switch :
-  (bool t * unit t) list ->
-  default:unit t -> unit t
+  [ `Case of bool t * unit t | `Default of unit t ] list -> unit t
+(** Create a switch statement from a list of {!case} and optionally a
```

Comment:
I'm curious why you chose to expose this type of interface (with the exception failure) as opposed to exporting an option argument for the `default` case?